File: source-connector/src/main/java/sourceconnector/domain/processor/impl/TrimMapperProcessor.java
package sourceconnector.domain.processor.impl;

import lombok.RequiredArgsConstructor;
import sourceconnector.domain.log.factory.LogFactory;
import sourceconnector.domain.log.Log;
import sourceconnector.domain.processor.AbstractMapperProcessor;

/**
 * Remove the whitespace in the Log
 */
@RequiredArgsConstructor
public class TrimMapperProcessor extends AbstractMapperProcessor<Log> {
  private final LogFactory logFactory;

  @Override
  protected Log map(Log log) {
    String trimmed = log.get().trim();
    return logFactory.create(trimmed, log.getMetadata());
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/processor/impl/TrimMapperProcessor.kt
package sourceconnector.domain.processor.impl

import sourceconnector.domain.log.Log
import sourceconnector.domain.log.factory.LogFactory
import sourceconnector.domain.processor.AbstractMapperProcessor

/**
 * Remove the whitespace in the Log
 */
class TrimMapperProcessor(
  private val logFactory: LogFactory
) : AbstractMapperProcessor<Log>() {

  override fun map(input: Log): Log {
    val trimmed = input.get().trim { it <= ' ' }
    return logFactory.create(trimmed, input.metadata)
  }
}
